src/components/network-map-tab.js

```diff
@@ -194,78 +183,123 @@ export const NetworkMapTab = ({
+    );
+
     const reloadMapGeoData = useCallback(() => {
-        console.info(`Loading geo data of study '${studyUuid}'...`);
-        const substationPositions = fetchSubstationPositions(
-            studyUuid,
-            currentNode?.id
-        );
-        const linePositions = lineFullPath
-            ? fetchLinePositions(studyUuid, currentNode?.id)
-            : [];
-        setWaitingLoadGeoData(true);
-
-        Promise.all([substationPositions, linePositions])
-            .then((values) => {
-                const newGeoData = new GeoData();
-                newGeoData.setSubstationPositions(values[0]);
-                newGeoData.setLinePositions(values[1]);
-                setGeoData(newGeoData);
-                setWaitingLoadGeoData(false);
-            })
-            .catch(function (error) {
-                console.error(error.message);
-                setWaitingLoadGeoData(false);
-                setErrorMessage(
-                    intlRef.current.formatMessage(
-                        { id: 'geoDataLoadingFail' },
-                        { studyUuid: studyUuid }
-                    )
+        if (studyUuid && currentNode) {
+            if (
+                geoData &&
+                (geoData.substationPositionsById.size > 0 ||
+                    geoData.linePositionsById.size > 0)
+            ) {
+                const missingSubstationPositions =
+                    getMissingEquipmentsPositions(
+                        geoData.substationPositionsById,
+                        network.getSubstations(),
+                        fetchSubstationPositionsByIds
+                    );
+
+                const missingLinesPositions = getMissingEquipmentsPositions(
+                    geoData.linePositionsById,
+                    network.getLines(),
+                    fetchLinePositionsByIds
                 );
-            });
-        dispatch(resetMapReloaded());
+
+                Promise.all([
+                    missingSubstationPositions,
+                    missingLinesPositions,
+                ]).then((positions) => {
+                    geoData.addSubstationPositions(positions[0]);
+                    geoData.addLinePositions(positions[1]);
+                    // If there is new substation positions, we instantiate a new Map so that the substations layer is rendering is triggered.
+                    // Same for line positions.
+                    const newGeoData = new GeoData(
+                        positions[0].length > 0
+                            ? new Map(geoData.substationPositionsById)
+                            : geoData.substationPositionsById,
+                        positions[1].length > 0
+                            ? new Map(geoData.linePositionsById)
+                            : geoData.linePositionsById
+                    );
+                    setGeoData(newGeoData);
+                    setWaitingLoadGeoData(false);
+                });
```

Comment:
Manage a catch here at least to print a message and `setWaitingLoadGeoData(false);`